### PR TITLE
Properly fix T/F lint

### PR DESCRIPTION
**Pull request by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q**

### CODE REVIEW

File: R/autoFormat.R [hunk 1]
```diff
@@ -5,6 +5,8 @@
 #' @param files A character vector of paths to files that should be auto-formatted.
 #' @param ignoreLintFreeFiles If set to TRUE (the default) files without linter warnings are not auto-formatted.
 #' @param lintAfterwards If set to TRUE (the default) return linter results for the auto-formatted files.
+#' @param askTF Ask about whether to replace T/F with TRUE/FALSE, or just do it
+#'     (default).
```
Comment (mikapfl) [inline R/autoFormat.R:9]:
```suggestion
```

File: R/autoFormat.R [hunk 2]
```diff
@@ -14,34 +16,17 @@
 #' lucode2::autoFormat()
 #' }
 #' @export
-autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE, lintAfterwards = TRUE) {
+autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE,
+                       lintAfterwards = TRUE, askTF = FALSE) {
```
Comment (mikapfl) [inline R/autoFormat.R:20]:
```suggestion
autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE,
                       lintAfterwards = TRUE) {
```

File: R/autoFormat.R [hunk 3]
```diff
@@ -14,34 +16,17 @@
...
+
+  # fix T_and_F_symbol lint (will only touch files with that kind of lint)
+  theOneAndonlyTandFsymbolFixer(files, askTF)
```
Comment (mikapfl) [inline R/autoFormat.R:23]:
```suggestion
  theOneAndonlyTandFsymbolFixer(files)
```

File: R/autoFormat.R [hunk 4]
```diff
@@ -51,3 +36,57 @@ autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE, lin
     print(lint(files))
   }
 }
+
+# Fix all instances of T/F rather than TRUE/FALSE indicated by
+# lintr::T_and_F_symbol_linter().
+theOneAndonlyTandFsymbolFixer <- function(files, ask = FALSE) {
```
Comment (mikapfl) [inline R/autoFormat.R:42]:
```suggestion
theOneAndonlyTandFsymbolFixer <- function(files) {
```

File: R/autoFormat.R [hunk 5]
```diff
@@ -51,3 +36,57 @@ autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE, lin
...
+        # if told to ask, ask about replacing or not
+        if (ask) {
+          message(paste("Replace", t[l], "with", u,
+                        paste0("in ", f[[1]][["filename"]], "? (Y/n)"),
+                        sep = "\n"))
+          # if not "yes, please", go to next occurrence
+          if (!tolower(getLine()) %in% c("", "y", "yes")) {
+            next
+          }
+        }
```
Comment (mikapfl) [inline R/autoFormat.R:79]:
```suggestion
```

File: R/autoFormat.R [hunk 6]
```diff
@@ -51,3 +36,57 @@ autoFormat <- function(files = getFilesToLint(), ignoreLintFreeFiles = TRUE, lin
...
+
+        # if not told to ask, or told to replace, replace
```
Comment (mikapfl) [inline R/autoFormat.R:81]:
```suggestion
```

**Review by mikapfl — APPROVED** [review]
Nice! I think it doesn't need to be optional, so I propose to take out the `askTF` thing, but I'm fine either way.

**Review by pfuehrlich-pik — APPROVED** [review]

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
> [pfuehrlich-pik](https://github.com/pfuehrlich-pik) approved these changes Dec 8, 2022

With or without `askTF`?

**Review by mikapfl — APPROVED** [review]

**Comment by pfuehrlich-pik** [comment]
I just got
```
> lucode2::autoFormat('R/foodmodelstat.R')
Error in `f[[1]]`:
! subscript out of bounds
Backtrace:
 1. lucode2::autoFormat("R/foodmodelstat.R")
 2. lucode2:::theOneAndOnlyTandFsymbolFixer(files)
 3. base::readLines(f[[1]][["filename"]])
```
Could you take a look @0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q ? I was in `/p/tmp/pascalfu/magpie4`

**Comment by 0UmfHxcvx5J7JoaOhFSs5mncnisTJJ6q** [comment]
See #145

**Pull request merged**